### DISCUSSION TITLE
del

### DISCUSSION THREAD

**Discussion by ViktorsBaikers** [openzfs/zfs#15119] [Q&A]
del

**Comment by oller22** [comment]
You have Raid0!!!
You can't delete the disk!
You could have replaced the disk!

now
The low speed is due to the HBA\Raid controller, it is not coping. 
There is a risk, but if you spread the disks over 2-4 controllers, the speed will increase many times over


But once again, you have Raid0! What your array is doing now is unclear to me


Pardon my poor English

**Comment by filip-paczynski** [comment]
Well, this is an interesting zpool layout, to say the least.

If I were to be faced with this:
1. I'd mount it read-only and copy whatever is still accessible
   - one would *probably* need to use "copy and ignore read errors" setting
   - I would not use `zfs send` for this
2. I'd monitor the faulted device with `iostat` or similar tool to see if it is in fact doing anything (the transfer rate reported by `zpool status` might be from the past)
3. When the copying (1.) is finished, then I'd maybe look into how to bring the faulted dev back and make zfs see it
4. Maybe then ZFS will finish removing it, maybe not
5. If it does copy again and re-design the pool with any kind of redundancy

This kind of layout should only be used for data that one does not care about.
If, instead of ZFS, any kind of raid tool was to be used, I think It would just give up immediately.

**Reply by filip-paczynski** [reply]
It's kind of amazing that getting anything from there is still possible - no matter how slowly.
If the faulted dev is still plugged in, then it might be saturating the controller with errors/re-reads/resets... - but I am not a SCSI/HDD firmware expert, so I'm most likely wrong.

Don't mess with it too much, just copy whatever is still accessible.

**Reply by filip-paczynski** [reply]
I doubt anyone here could help You with that.
There is no redundancy in this pool, the data stored on the faulted device is lost (currently).

This scenario is somewhat dangerous, but... If this is a machine to which You have a physical access: shut it down, attach a clean HDD to it (leve the old one), reboot into rescue mode (no ZFS), `dd` the faulted device to clean drive, replace the faulted dev with the new one, see what happens.

Why dangerous? Because I do not known how exactly ZFS handles faulted dev on stripe(raid0) with no redundancy + device change.
Might as well loose everything.
